src/ZkFold/Symbolic/Algorithms/Hash/Blake2b.hs

```diff
@@ -109,14 +98,15 @@ blake2b_compress Blake2bCtx{h, m, t} lastBlock =
         v1 = V.foldl hashRound v0 $ fromList [0..11] -- twelve rounds
     in fmap (\(i, hi) -> hi `xorUInt` (v1 ! i) `xorUInt` (v1 ! (i GHC.+ 8))) (V.zip (fromList [0..7]) h)
 
-blake2b' :: forall bb' kk' ll' nn' b .
-    ( KnownNat bb'
+blake2b' :: forall bb' kk' ll' nn' c .
+    ( Symbolic c
+    , KnownNat bb'
```

Comment:
Are these parameters used on the type level? Maybe we can move them to the value level